compiler/rustc_codegen_ssa/src/base.rs

```diff
@@ -891,14 +911,11 @@ impl CrateInfo {
             let missing =
                 missing.iter().cloned().filter(|&l| lang_items::required(tcx, l)).collect();
             info.missing_lang_items.insert(cnum, missing);
+        }
 
-            // Only include debugger visualizer files from crates that will be statically linked.
-            if used_crate_source.rlib.is_some() || used_crate_source.rmeta.is_some() {
-                let debugger_visualizers = tcx.debugger_visualizers(cnum).clone();
-                if !debugger_visualizers.is_empty() {
-                    info.debugger_visualizers.insert(cnum, debugger_visualizers);
-                }
-            }
+        if tcx.sess.target.is_like_msvc {
+            info.debugger_visualizers =
```

Comment:
Worth renaming to `natvis_debugger_visualizers` to make it clear we only care for those in terms of what the linker is given later?

Reply:
This sounds good, I've renamed the field to express it's only accounting for NatVis debugger visualizers.